packages/gatsby-recipes/src/providers/gatsby/plugin.js

```diff
@@ -376,9 +454,29 @@ class BabelPluginGetPluginsFromGatsbyConfig {
               return
             }
 
+            // Only allow module.exports = {}
+            // Not things like module.exports = () => ({})
+            if (!t.isObjectExpression(right)) {
+              return
+            }
+
             const plugins = right.properties.find(p => p.key.name === `plugins`)
 
-            plugins.value.elements.map(node => {
+            let pluginsList = []
+
+            if (t.isCallExpression(plugins.value)) {
+              pluginsList = plugins?.value?.callee?.object?.elements
```

Comment:
You have already validated that `plugins.value` is not nullable. Additionally, you asserted that it is a `CallExpression`, so the `callee` property must exist as well and it cannot be nullable either.
```suggestion
              pluginsList = plugins.value.callee.object?.elements
```